source/common/router/req_header_formatter.cc

```diff
@@ -0,0 +1,81 @@
+#include "common/router/req_header_formatter.h"
+
+#include <cstdint>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "common/config/rds_json.h"
+#include "common/http/headers.h"
+#include "common/json/json_loader.h"
+
+namespace Envoy {
+namespace Router {
+
+HeaderFormatterPtr RequestHeaderParser::parseInternal(const std::string& format) {
+  if (format.find("%") == 0) {
+    const size_t last_occ_pos = format.rfind("%");
+    if (last_occ_pos == std::string::npos || last_occ_pos <= 1) {
+      throw EnvoyException(fmt::format(
+          "Incorrect configuration: {}. Expected the variable to be of format %<variable_name>%",
+          format));
+    }
+    const std::string variable_name = format.substr(1, last_occ_pos - 1);
+    return HeaderFormatterPtr{new RequestHeaderFormatter(variable_name)};
+  } else {
+    return HeaderFormatterPtr{new PlainHeaderFormatter(format)};
+  }
+}
+
+RequestHeaderParserPtr RequestHeaderParser::parse(
+    const Protobuf::RepeatedPtrField<envoy::api::v2::HeaderValueOption>& headers) {
+  RequestHeaderParserPtr request_header_parser(new RequestHeaderParser());
+  for (const auto& header_value_option : headers) {
+    ENVOY_LOG(debug, "adding key {} to header formatter map", header_value_option.header().key());
+    HeaderFormatterPtr header_formatter =
+        RequestHeaderParser::parseInternal(header_value_option.header().value());
+    request_header_parser->header_formatter_map_.emplace(
+        Http::LowerCaseString(header_value_option.header().key()), std::move(header_formatter));
+  }
+
+  return request_header_parser;
+}
+
+void RequestHeaderParser::evaluateRequestHeaders(
+    Http::HeaderMap& headers, const Http::AccessLog::RequestInfo& requestInfo,
+    const std::list<std::pair<Http::LowerCaseString, std::string>>& requestHeadersToAdd) const {
```

Comment:
request_headers_to_add (please go through your entire diff and look for snake_case issues. Thanks)